File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let ret = []
    const max = Math.max(...candies)

    candies.forEach((candy) => {
        ret.push((candy+extraCandies) >= max);
    })

    return ret;
};

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> ret = new ArrayList<>();
        
        int max = Arrays.stream(candies).max().getAsInt();

        for (int i=0; i<candies.length; i++) {
            ret.add(candies[i]+extraCandies >= max);
        }
        return ret;
    }
}